Dockerfile

```diff
@@ -69,76 +39,73 @@ ENV CMS_CFG="$CONFIG_ROOT/cms.yml"
-# via pip.
-# The node environment is already 'activated' because its .../bin was put on $PATH.
-RUN nodeenv /edx/app/edxapp/nodeenv --node=16.14.0 --prebuilt
-RUN npm install -g npm@8.5.x
-COPY package.json package.json
-COPY package-lock.json package-lock.json
-RUN npm set progress=false && npm install
-
-# Copy over remaining parts of repository (including all code).
 COPY . .
 
-# Install Python requirements again in order to capture local projects
-RUN pip install -e .
-
+# Create user before assigning any directory ownership to it.
 RUN useradd -m --shell /bin/false app
 
-USER app
+RUN set -eux; \
+    apt-get update; \
+    apt-get -y dist-upgrade; \
+    echo "locales locales/default_environment_locale select en_US.UTF-8" | debconf-set-selections; \
+    echo "locales locales/locales_to_be_generated multiselect en_US.UTF-8 UTF-8" | debconf-set-selections; \
+    apt-get --yes install --no-install-recommends ${APT_MANDATORY_REQUIREMENTS}; \
+    savedAptMark="$(apt-mark showmanual)"; \
+    apt-mark auto '.*' > /dev/null; \
+    apt-mark manual $savedAptMark > /dev/null; \
+    apt-get purge -y --auto-remove -o APT::AutoRemove::RecommendsImportant=false; \
+    savedAptMark="$(apt-mark showmanual)"; \
+    apt-get --yes install --no-install-recommends ${APT_BUILD_REQUIREMENTS}; \
+    # Set up a Python virtual environment.
+    # It is already 'activated' because $VIRTUAL_ENV/bin was put on $PATH.
+    python3.8 -m venv "${VIRTUAL_ENV}"; \
+    pip install -r requirements/pip.txt; \
+    pip install -r requirements/edx/base.txt; \
+    # Install development requirements only If DEVELOPMENT argument is set to true at build time.
+    # Link configuration YAMLs and set EDX_PLATFORM_SETTINGS.
+    if $DEVELOPMENT ; then \
+        pip install -r requirements/edx/development.txt && \
+        ln -s "$(pwd)/lms/envs/devstack-experimental.yml" "$LMS_CFG" && \
+        ln -s "$(pwd)/cms/envs/devstack-experimental.yml" "$CMS_CFG" && \
+        # Temporary compatibility hack while devstack is supporting
+        # both the old `edxops/edxapp` image and this image:
+        # Add in a dummy ../edxapp_env file.
+        # The edxapp_env file was originally needed for sourcing to get
+        # environment variables like LMS_CFG, but now we just set
+        # those variables right in the Dockerfile.
+        touch ../edxapp_env; \
+    fi; \
+    # Set up a Node environment and install Node requirements.
+    # Must be done after Python requirements, since nodeenv is installed
+    # via pip.
+    # The node environment is already 'activated' because its .../bin was put on $PATH.
+    nodeenv /edx/app/edxapp/nodeenv --node=16.14.0 --prebuilt; \
+    npm install -g npm@8.5.x; \
+    npm set progress=false && npm install; \
+    # Install Python requirements again in order to capture local projects
+    pip install -e .; \
+    pip cache purge; \
+    apt-get --yes remove --purge ${APT_BUILD_REQUIREMENTS}; \
```

Comment:
Can you leave some comments about why apt should be purged of the build requirements and `/var/lib/apt/*` needs to be deleted?